### DISCUSSION TITLE
error with value key on "usePrepareContractWrite" - works locally, fails on vercel

### DISCUSSION THREAD

**Discussion by jcislo** [wevm/wagmi#2504] [Q&A]
This is my current usePrepareContractWrite hook:

  I had the issue with value being undefined at first, but managed to fix that by replacing the ABI with just the single function snippet. Also tried some other things like casting (as was recommended in another post). This works locally, however the error persists on deployment to Vercel where the message states "Value" is not a member of "usePrepareContractWriteConfig"

```const { config } = usePrepareContractWrite({
    address: `0x${Book.discoveryMinter}`,
    abi: [
      {
        inputs: [
          {
            internalType: "uint256",
            name: "amt",
            type: "uint256"
          }
        ],
        name: "buyKey",
        outputs: [],
        stateMutability: "payable",
        type: "function",
        payable: true
      },
    ],
    functionName: 'buyKey',
    args: [text],
    enabled: Boolean(text),
    value: ethers.parseEther((0.01 * text).toString()) as any
  })

**Comment by ombayoeco11** [comment]
is it ethers 6? try the viem then...
import { parseEther } from 'viem' ;

i dunno if the value for payable is like that. perhaps if you set the value : parseEther("0.01",18) should be enough?
